README.md

```diff
@@ -6,12 +6,244 @@
+of key-value cache storages. The basic interface includes methods to add, get
+and remove objects by key.
+- `Cache` class to create a strict cache storage by a given name for a specified
+`Cachable`-compliant type.
+- `HybridCache` class that works with every kind of `Cachable`-compliant types.
+- Flexible `Config` struct which is used in the initialization of `Cache` and
+`HybridCache` classes, based on the concept of having front- and back- caches.
+A request to a front cache should be less time and memory consuming (`NSCache` is used
+by default here). The difference between front and back caching is that back
+caching is used for content that outlives the application life-cycle. See it more
+like a convenient way to store user information that should persist across application
+launches. Disk cache is the most reliable choice here.
+- `StorageFactory` - a place to register and retrieve your cache storage by type.
+- Possibility to set expiry date + automatic cleanup of expired objects.
+- Basic memory and disk cache functionality.
+- Scalability, you are free to add as many cache storages as you want
+(if default implementations of memory and disk caches don't fit your purpose for some reason).
+- `NSData` encoding and decoding required by `Cachable` protocol are implemented
+for `UIImage`, `String`, `JSON` and `NSData`.
+
 ## Usage
 
+### Hybrid cache
+
+`HybridCache` supports storing all kinds of objects, as long as they conform to
+the `Cachable` protocol. It's two layered cache (with front and back storages),
+as well as `Cache`.
+
+**Initialization with default configuration**
+
 ```swift
-<API>
+let cache = HybridCache(name: "Mix")
+```
+
+**Initialization with custom configuration**
+
+```swift
+let config = Config(
+  // Your front cache type
+  frontKind: .Memory,
+  // Your back cache type
+  backKind: .Disk,
+  // Expiry date that will be applied by default for every added object
+  // if it's not overridden in the add(key: object: expiry: completion:) method
+  expiry: .Date(NSDate().dateByAddingTimeInterval(100000)),
+  // Maximum size of your cache storage    
+  maxSize: 10000)
+
+let cache = HybridCache(name: "Custom", config: config)
+```
+
+**Basic operations**
+
+```swift
+let cache = HybridCache(name: "Mix")
+
+// String
+cache.add("string", object: "This is a string")
+
+cache.object("string") { (string: String?) in
+  print(string) // Prints "This is a string"
+}
+
+// JSON
+cache.add("jsonDictionary", object: JSON.Dictionary(["key": "value"]))
+
+cache.object("jsonDictionary") { (json: JSON?) in
+  print(json?.object)
+}
+
+// UIImage
+cache.add("image", object: UIImage(named: "image.png"))
+
+cache.object("image") { (image: UIImage?) in
+  // Use your image
+}
+
+// NSData
+cache.add("data", object: data)
+
+cache.object("data") { (data: NSData?) in
+  // Use your NSData object
+}
+
+// Remove an object from the cache
+cache.remove("data")
+
+// Clean the cache
+
+cache.clear()
 ```
 
+### Strict cache
```

Comment:
Well, then I guess no one would use it ^^
But I think the other one makes sense, it's hybrid.